Improve docstrings for interpolator/regridder schemes

dkillick opened the pull request:
Improves the docstrings for the interpolator and regridder schemes `Linear`, `AreaWeighted` and `Nearest` in `iris.analysis`. This adds:
 * the word "regridding" into the docstrings where these schemes have been extended to be used as a regridding scheme as well as an interpolation scheme,
 * usage examples to each schemes' `regridder` method, and
 * a comment on typical usage of the `callable` returned by each scheme's `interpolator` and `regridder` methods to the docstrings of these methods.

rhattersley commented on lib/iris/analysis/__init__.py:
```diff
@@ -1597,7 +1601,13 @@ def regridder(self, src_grid, target_grid):
+            be on the same grid as src_grid.
+
+            This callable would typically be used by
+            :class:`iris.cube.Cube.regrid`.
```
:+1: for greater clarity in the description of the `regridder()` method and its resulting callable. However, I don't think the "Thus for the callable ..." statement helps. And the "This callable would typically ..." statement risks making matters worse. (e.g. Am I supposed to pass the result of Linear().regridder(..., ...) to cube.regrid()?)

How about:
 - Leaving the "Returns:" section as it was prior to this PR (other than the "which" -> "that" substitution).
 - Adding more detail above the "Args:" section to explain that the first port of call is to use Cube.regrid() and not use this method at all. But when doing repeated regridding operations and cube.regrid() is too inefficient then this section can demonstrate an example usage pattern. As a rough and ready example:
```python
regridder = Linear().regridder(src_grid, target_grid)
for cube in my_long_list_of_cubes_on_the_src_grid:
    cube_on_target_grid = regridder(cube)
    # Do stuff with cube_on_target_grid ...
```

dkillick replied:
:+1:

bjlittle replied:
:+1: Thanks @dkillick 

> regridder = Linear().regridder(src_grid, target_grid) for cube in my_long_list_of_cubes_on_the_src_grid: cube_on_target_grid = regridder(cube) # Do stuff with cube_on_target_grid ...

I'd second highlighting the above use pattern in the doc-string as it may not be immediately obvious to the user.

dkillick replied:
> I'd second highlighting the above use pattern in the doc-string 

:+1: but actually I think I'm going to replace it with a link to a section in the user guide chapter on interpolate and regrid. This section (that doesn't currently exist!) will include example code very similar to the above but will be a little longer and give more context.

Doing this will prevent duplicating this precise example in every `regridder` docstring for each regridding scheme :scream_cat:

dkillick commented:
Here's some further changes to the interpolator/regridder schemes docstrings. I've...
 * removed the complicated description of how to set up your own regridder/interpolator,
 * replaced the above with links into the interpolation and regridding userguide chapter, and
 * added a couple of references to the userguide chapter to allow the links described above.

bjlittle commented:
@dkillick Okay, I'm on it ...

bjlittle commented:
:+1: Lovely, thanks @dkillick
